In [5]:
# importons les bibliothèques 
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

from sklearn.model_selection import train_test_split

from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline

from sklearn.preprocessing import StandardScaler, OneHotEncoder

from sklearn.linear_model import LinearRegression, LogisticRegression
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import mean_squared_error, accuracy_score, classification_report

from imblearn.over_sampling import SMOTE




In [15]:
# Chargeons nos données
df = pd.read_csv("../data/raw/healthcare-dataset-stroke-data.csv")

print(df.head())

      id  gender   age  ...   bmi   smoking_status stroke
0   9046    Male  67.0  ...  36.6  formerly smoked      1
1  51676  Female  61.0  ...   NaN     never smoked      1
2  31112    Male  80.0  ...  32.5     never smoked      1
3  60182  Female  49.0  ...  34.4           smokes      1
4   1665  Female  79.0  ...  24.0     never smoked      1

[5 rows x 12 columns]


In [16]:
# nous allons faire une copie de notre dataset pour éviter de l'endommager
df1 = df.copy()

In [17]:
# nous allons supprimer la colonne "id" qui n'est pas un paramètre médical ici, mais d'identification des patients de manière anonyme
df1 = df1.drop("id", axis=1)

print(df1.head())

   gender   age  hypertension  ...   bmi   smoking_status stroke
0    Male  67.0             0  ...  36.6  formerly smoked      1
1  Female  61.0             0  ...   NaN     never smoked      1
2    Male  80.0             0  ...  32.5     never smoked      1
3  Female  49.0             0  ...  34.4           smokes      1
4  Female  79.0             1  ...  24.0     never smoked      1

[5 rows x 11 columns]


## Feature Engineering (Choix des variables à conserver)
### Création des variables X et y ( Sélectionne que les variables numériques)

In [ ]:
# Défissons les variables X et y
X=df1.drop(columns="stroke") # variables explicatives
y= df1["stroke"]                # variable cible


Les caractéristiques (X) du patient nous permettrons de prédire la présence ou non d'un AVC (y) à l'aide d'un algorithme (Modèle du Machine Learning).

In [ ]:
# vérifions
print(f" X se représente ainsi: \n {X.head()}")

print()

print(f" y se présente ainsi: \n {y.head()}")

 X se représente ainsi: 
    gender   age  hypertension  ...  avg_glucose_level   bmi   smoking_status
0    Male  67.0             0  ...             228.69  36.6  formerly smoked
1  Female  61.0             0  ...             202.21   NaN     never smoked
2    Male  80.0             0  ...             105.92  32.5     never smoked
3  Female  49.0             0  ...             171.23  34.4           smokes
4  Female  79.0             1  ...             174.12  24.0     never smoked

[5 rows x 10 columns]

 y se présente ainsi: 
 0    1
1    1
2    1
3    1
4    1
Name: stroke, dtype: int64


In [26]:
print(f"X est composé de: {X.shape[0]} lignes et {X.shape[1]} colonnes")
print(f"y est composé de: {y.shape[0]} lignes")

X est composé de: 5110 lignes et 10 colonnes
y est composé de: 5110 lignes


y = 5110 est une série, un seul vecteur de 5 110 valeurs, ce qui est exactement ce qu'attendent les modèles de scikit-learn en tant que variable cible.

### Séparer les jeux d'entraînement et les jeux de test (jeu de test à 20% et random_state = 42)

### Choisir et importer un algorithme de ML (Algorithme de Machine Learning)